lib/validators/specResolver.js

```diff
@@ -657,9 +657,32 @@ class SpecResolver {
     // Adding the model name or it's discriminator value as an enum constraint with one value (constant) on property marked as discriminator
     if (definitions[name] && definitions[name].properties && definitions[name].properties[discriminator]) {
       let val = name;
+      let discriminatorType = 'string';
       if (definitions[name]['x-ms-discriminator-value']) {
         val = definitions[name]['x-ms-discriminator-value'];
       }
+      // Ensure that the property marked as a discriminator has only one enum constraint and it should be the one that we want to set as a constant.
+      if (definitions[name].properties[discriminator].enum) {
```

Comment:
it seems that we figure out the values the discriminator should take and add them all in the polymorphic tree, therefore we'd like to ignore whatever the spec may indicate on this "type" if there was an enum defined, is this correct? If so, could we make a comment to indicate this as well?

Reply:
yes that is correct. I will update the above comment to be more clear.